Added static setting for checkPendingFlushUpdate functionality of lucene index writer (#6728)
opensearch-project/documentation-website: _install-and-configure/configuring-opensearch/index-settings.md

- `index.merge_on_flush.policy` (default | merge-on-flush): This setting controls which merge policy should be used when `index.merge_on_flush.enabled` is enabled. Default is `default`.

> - `index.check_pending_flush.enabled` (Boolean): This setting controls Apache Lucene's Index writer setting checkPendingFlushOnUpdate which controls if an indexing thread should check for pending flushes on update in order to help out flushing indexing buffers to disk

Reviewer: ```suggestion
- `index.check_pending_flush.enabled` (Boolean): This setting controls the Apache Lucene's `checkPendingFlushOnUpdate` index writer setting, which specifies whether an indexing thread should check for pending flushes on an update in order to flush indexing buffers to disk.
```

Reviewer: Is there a default for this setting?

Author: Yes Default is `true`

> - `index.check_pending_flush.enabled` (Boolean): This setting controls the Apache Lucene's `checkPendingFlushOnUpdate` index writer setting, which specifies whether an indexing thread should check for pending flushes on an update in order to flush indexing buffers to disk. Default is `true`.

Reviewer 2: ```suggestion
- `index.check_pending_flush.enabled` (Boolean): This setting controls the Apache Lucene `checkPendingFlushOnUpdate` index writer setting, which specifies whether an indexing thread should check for pending flushes on an update in order to flush indexing buffers to disk. Default is `true`.
```
